src/graph/transform/compact.cc

```diff
@@ -36,13 +36,30 @@ CompactGraphs(
   std::vector<aten::IdHashMap<IdType>> hashmaps(num_ntypes);
   std::vector<std::vector<EdgeArray>> all_edges(graphs.size());   // all_edges[i][etype]
 
-  for (size_t i = 0; i < always_preserve.size(); ++i)
+  std::vector<int64_t> max_vertex_cnt(num_ntypes, 0);
+  for (size_t i = 0; i < graphs.size(); ++i) {
+    const HeteroGraphPtr curr_graph = graphs[i];
+    for (IdType ntype = 0; ntype < num_ntypes; ++ntype) {
+      max_vertex_cnt[ntype] += curr_graph->NumVertices(ntype);
```

Comment:
`CompactGraphs` compacts multiple graphs with the same node space, so there is no need to repeatedly add `NumVertices()`.

Also, note that `CompactGraphs` operates on the original node space (meaning that the number of nodes would be the same as full graph).  Would reserving that many nodes cost a lot of memory?

Reply:
I have updated to get NumEdges instead of NumVertices and the numbers above are updated